challenges/validate.ts

```diff
@@ -91,4 +148,142 @@ const validateMetadataFiles = () => {
   challengeIds.forEach(validatePrerequisiteIds);
 };
 
-validateMetadataFiles();
+const validateTests = () => {
+  getChallengeIds()
+    .filter((id) => {
+      const path = join(challengesDir, id, 'solutions');
+      statSync(path).isDirectory();
+      readdirSync(path).forEach((file) => {
+        if (!/^\d+\.ts$/.test(file)) {
+          throw new Error(
+            `the solution file name: ${join(path, file)} is invalid: it must only be numeric.`,
+          );
+        }
+      });
+      return true;
+    })
+    .forEach((id) => {
+      if (!isFocused(id)) {
+        return;
+      }
+
+      const compilerOptions = readConfigFile(join(challengesDir, id, 'tsconfig.json'), sys.readFile)
+        .config.compilerOptions as CompilerOptions;
+
+      const testsSource = readFileSync(join(challengesDir, id, 'tests.ts'), 'utf8');
+
+      readdirSync(join(challengesDir, id, 'solutions')).forEach((file) => {
+        const solutionPath = join(challengesDir, id, 'solutions', file);
+        const solutionSource = readFileSync(solutionPath, 'utf8');
+
+        // our challenge test file
+        const sourceFile = createSourceFile(
+          `${repoRoot}/in-memory/${file}`,
```

Comment:
in the future we should split them up into two files `user.ts` and `test.ts` would help narrowing down the error lines in output easier.

I was playing with that here
https://github.com/Hacksore/typecheck-service/blob/feat/worker-cf/src/server.ts#L52-L53

Reply:
I was thinking that too, but the problem there is we'd need to insert our own imports which will throw off line counts.  beyond that, for imports we'd need to export things which is probably a bigger issue because then we'd need to go in and find all the symbols the user may have created.  or am I missing something?

Reply:
agreed it's not required now let me resolve this.